File: 6 kyu/Pair of gloves/solution.py
def number_of_pairs(gloves):
    dct = {el:gloves.count(el) for el in gloves}
    return sum([el//2 for el in list(dct.values())])
  
_____________________________
from collections import Counter

def number_of_pairs(gloves):
    return sum(c // 2 for c in Counter(gloves).values())
  
______________________
def number_of_pairs(gloves):
    return sum(gloves.count(color)//2 for color in set(gloves))
  
____________________
def number_of_pairs(gloves):
    unique = set(gloves)
    return sum(gloves.count(i)//2 for i in unique)
  
____________________
from collections import Counter

def number_of_pairs(gloves):
    return sum(num // 2 for num in Counter(gloves).values())
  
____________________
def number_of_pairs(gloves):
    return 0 if len(gloves) < 2 else sum([gloves.count(x) // 2 for x in set(gloves)])
  
__________________
def number_of_pairs(gloves):
    d={}
    for i in gloves:
        if i in d:
            d[i]+=1
        else:
            d[i]=1
    return sum([i//2 for d,i in d.items()])
  
_______________
def number_of_pairs(gloves):
    a_dict = {}
    for colour in gloves:
        if not colour in a_dict:
            a_dict[colour] = 1
        else:
            a_dict[colour] += 1
    pairs = 0
    for value in a_dict.values():
        pairs += value//2
    return pairs

File: 6 kyu/Pair of gloves/solution.rb
def number_of_pairs(gloves)
  gloves.uniq.map{|color|gloves.count(color)/2}.inject 0,:+
end

__________________
def number_of_pairs(g)
  g.group_by(&:itself).transform_values(&:size).values.map {|c| c / 2}.sum
end

___________
def number_of_pairs(gloves)
  gloves.each_with_object(Hash.new(0)) { |c, frq| frq[c] += 1 }.values.reduce(0) { |s, v|s += v/2 }
end

_______________
def number_of_pairs(gloves)
  gloves.group_by(&:itself).sum{|_,arr| arr.size/2}
end

_______________
def number_of_pairs(gloves)
  gloves.uniq.map{|e| gloves.count(e)/2}.sum
end

______________
def number_of_pairs(gloves)
  gloves.uniq.map { |glove| gloves.count(glove) / 2 }.sum
end

_____________
def number_of_pairs(gloves)
  gloves.uniq.inject(0) { |memo, color| memo + gloves.count(color) / 2 }
end

_____________
def number_of_pairs(gloves)
  pair = 0
  gloves.uniq.each{ |glove| pair += gloves.count(glove)/2 }
  return pair
end

_____________
def number_of_pairs(gloves)
  @gloves = gloves
  @gloves = (@gloves.uniq.map!{ |x| gloves.count(x) / 2 }).inject(:+)
  @gloves.to_i > 0 ? @gloves :  0
end
